core/kernel/dt.c

```diff
@@ -53,6 +57,51 @@ const struct dt_driver *__dt_driver_end(void)
 	return &__rodata_dtdrv_end;
 }
 
+int dt_map_dev(const void *fdt, int offs, vaddr_t *base, size_t *size)
+{
+	enum teecore_memtypes mtype;
+	paddr_t pbase;
+	vaddr_t vbase;
+	ssize_t sz;
+	int st;
+
+	assert(cpu_mmu_enabled());
+
+	st = _fdt_get_status(fdt, offs);
+	if (st == DT_STATUS_DISABLED)
+		return -1;
+
+	pbase = _fdt_reg_base_address(fdt, offs);
+	if (pbase == (paddr_t)-1)
+		return -1;
+	sz = _fdt_reg_size(fdt, offs);
+	if (sz < 0)
+		return -1;
+
+	if ((st & DT_STATUS_OK_SEC) && !(st & DT_STATUS_OK_NSEC))
+		mtype = MEM_AREA_IO_SEC;
+	else
+		mtype = MEM_AREA_IO_NSEC;
+
+	vbase = (vaddr_t)phys_to_virt(pbase, mtype);
```

Comment:
Ah, yes, I assumed the memory would normally be mapped only through this function, which is obviously wrong. I'll fix.